### PR TITLE
Suggestion for implementation of clients API

### PR DESCRIPTION
**Pull request by cha0s**

### CODE REVIEW

**Comment by cha0s** [comment]
References https://github.com/Automattic/socket.io-adapter/pull/24

**Comment by rauchg** [comment]
This API is 100% spot on

File: Readme.md [hunk 1]
```diff
@@ -240,6 +240,18 @@ server.listen(3000);
   Hash of `Socket` objects that are connected to this namespace indexed
   by `id`.
 
+### Namespace#clients(fn:Function)
+
+  Gets a list of client IDs connected to this namespace (across all nodes if applicable).
+
+  ```js
+  var io = require('socket.io')();
+  io.of('/chat').in('general').clients(function(error, clients){
```
Comment (rauchg) [inline Readme.md]: I would also add an example of getting all rooms

**Comment by cha0s** [comment]
I added some examples to the docs

**Comment by barisusakli** [comment]
I assume, `io.in('general').clients();` will also work?

**Comment by cha0s** [comment]
Yeah, it reflects the broadcast API

**Comment by rauchg** [comment]
We need to rebase again

**Comment by cha0s** [comment]
I haven't done this for pull requests before. Do I just fix the history 
and --force push? Will that work or will it screw things up?

**Comment by rauchg** [comment]
Up to you. You can simply just merge from master and regular-push

**Comment by cha0s** [comment]
Well I tried rebasing off master but I didn't get any changes... So, I just force pushed to squash it into one commit. That cool?

**Comment by cha0s** [comment]
Nevermind, I forgot to sync my fork with upstream :) should be good now.

**Pull request merged**